### PR TITLE
Consolidate deployable entity tests

### PR DESCRIPTION
**Pull request by convoliution**
## Impact and Context

Client tests have various duplicated/overlapping tests between `ExperimentRun` and `RegisteredModelVersion`—the subclasses of `_DeployableEntity`. This PR consolidates those tests.

Note that tests in `deployable_entity/test_*.py` are each still specific to either `ExperimentRun` or `RegisteredModelVersion` (as they were just copy & pasted); #2604 makes the change to have those tests run with _both_ `ExperimentRun` and `RegisteredModelVersion`.

## Risks

Since tests are effectively being renamed, some test executions histories on Jenkins will lose their continuity.

## Changes

This PR is pure copy + paste + delete.

- add `deployable_entity/__init__.py`
- create `test_experimentrun/test_artifacts.py` from root `test_artifacts.py`
- create `deployable_entity/test_artifacts.py` from remaining root `test_artifacts.py`
- remove redundant artifact download tests from `registry/model_version/test_artifacts.py`
- remove redundant model artifact tests from `registry/model_version/test_deployment.py`
- create `test_utils/test_histogram_utils.py` from root `test_deployment.py`
- create `test_experimentrun/test_deployment.py` from root `test_deployment.py`
- create `deployable_entity/test_deployment.py` from remaining root `test_deployment.py` plus `registry/model_version/test_deployment.py`

## Testing

No new service, functionality, or tests added—this PR is just rearranging old tests.
- [ ] ~Deployed the service to dev env~
- [ ] ~Used functionality on dev env~
- [ ] ~Added unit test(s)~
- [ ] ~Added integration test(s)~

Verifying tests that are expected to still work:
- [x] `test_experimentrun/test_artifacts.py`
  - job/pytest-3.7/176 has one failure, addressed in #2605
- [x] `test_experimentrun/test_deployment.py`
  - job/pytest-3.7/177 all passed
- [x] `registry/model_version/test_artifacts.py`
  - job/pytest-3.7/178 all passed
- [x] `registry/model_version/test_deployment.py`
  - job/pytest-3.7/179 all passed

## How to Revert

Revert this PR

### CODE REVIEW

**Comment by conradoverta** [comment]
I will not review a PR with 1.2K LOC =)

**Comment by conradoverta** [comment]
Is this just code movement? Doesn't look like it from some removals.

File: client/verta/tests/registry/model_version/test_artifacts.py [hunk 1]
```diff
@@ -57,41 +57,6 @@ def test_add_artifact_file(self, model_version, in_tempdir, random_data):
...
-        assert new_filepath == os.path.abspath(new_filename)
-        with open(new_filepath, "rb") as f:
-            assert f.read() == FILE_CONTENTS
-
-        # object as well
-        obj = {"some": ["arbitrary", "object"]}
-        model_version.log_artifact(key, obj, overwrite=True)
-        new_filepath = model_version.download_artifact(key, new_filename)
-        with open(new_filepath, "rb") as f:
-            assert pickle.load(f) == obj
```
Comment (convoliution) [inline client/verta/tests/registry/model_version/test_artifacts.py:83]: Identical to https://github.com/VertaAI/modeldb/pull/2603/commits/a1810f318bfa3bcd579656ba46f9e110d6009280#diff-1a5838d629f2008e901066b48a5126a4c23f67ddf53f5dea2c69ab6433475ce8R202-R225

File: client/verta/tests/registry/model_version/test_artifacts.py [hunk 2]
```diff
@@ -57,41 +57,6 @@ def test_add_artifact_file(self, model_version, in_tempdir, random_data):
...
-
-    def test_download_directory(self, model_version, strs, dir_and_files, in_tempdir):
-        key, download_path = strs[:2]
-        dirpath, _ = dir_and_files
-
-        model_version.log_artifact(key, dirpath)
-        retrieved_path = model_version.download_artifact(key, download_path)
-
-        # contents match
-        utils.assert_dirs_match(dirpath, retrieved_path)
```
Comment (convoliution) [inline client/verta/tests/registry/model_version/test_artifacts.py:93]: Identical to https://github.com/VertaAI/modeldb/pull/2603/commits/a1810f318bfa3bcd579656ba46f9e110d6009280#diff-1a5838d629f2008e901066b48a5126a4c23f67ddf53f5dea2c69ab6433475ce8R227-R235

**Comment by convoliution** [comment]
This PR has been split into:
- #2612
- #2613
- #2614
- #2615
- #2616
- #2617
- #2618
- #2619
- #2620
- #2621
- #2622

As proof of accuracy, the diff between the above branches combined and this branch is only `test_log_path` moving positions:

```bash
$ git checkout ml/depl-entity-fixture-1 \
> && git checkout -b temp-consolidate-tests \
> && git merge \
> ml/init-grand \
> ml/consolidate-tests-1 \
> ml/consolidate-tests-2 \
> ml/consolidate-tests-3 \
> ml/consolidate-tests-4 \
> ml/consolidate-tests-5 \
> ml/consolidate-tests-6 \
> ml/consolidate-tests-7 \
> ml/consolidate-tests-8 \
> ml/cache-dir
$ git diff origin/ml/depl-entity-tests                                                                [0/1844]
diff --git a/client/verta/tests/test_experimentrun/test_artifacts.py b/client/verta/tests/test_experimentrun/test_artifacts.py
index 49225275f..884546713 100644
--- a/client/verta/tests/test_experimentrun/test_artifacts.py
+++ b/client/verta/tests/test_experimentrun/test_artifacts.py
@@ -9,6 +9,18 @@ from verta._internal_utils import _artifact_utils
  
  
 class TestArtifacts:
+    def test_log_path(self, experiment_run, strs):
+        strs, holdout = strs[:-1], strs[-1]  # reserve last key 
+
+        for key, artifact_path in zip(strs, strs):
+            experiment_run.log_artifact_path(key, artifact_path)
+
+        for key, artifact_path in zip(strs, strs):
+            assert experiment_run.get_artifact(key) == artifact_path
+
+        with pytest.raises(KeyError):
+            experiment_run.get_artifact(holdout)
+
     def test_clientside_storage(self, experiment_run, strs, in_tempdir, random_data):
         key = strs[0]
         filename = strs[1]
@@ -50,18 +62,6 @@ class TestArtifacts:
             else:
                 del os.environ[VERTA_ARTIFACT_DIR_KEY]
  
-    def test_log_path(self, experiment_run, strs):
-        strs, holdout = strs[:-1], strs[-1]  # reserve last key 
-
-        for key, artifact_path in zip(strs, strs):
-            experiment_run.log_artifact_path(key, artifact_path)
-
-        for key, artifact_path in zip(strs, strs):
-            assert experiment_run.get_artifact(key) == artifact_path
-
-        with pytest.raises(KeyError):
-            experiment_run.get_artifact(holdout)
-
     def test_download_path_only_error(self, experiment_run, strs, in_tempdir):
         key = strs[0]
         path = strs[1]
```

**Pull request closed without merging**